### PR TITLE
bump(main/vim,main/vim-python,x11/vim): 9.1.0700

### PR DESCRIPTION
**Pull request by TomJo2000**
I also deduplicated the patches using symlinks to the corresponding patch in `packages/vim`,
that way the same patch doesn't need to be updated in 3 separate places.

<details><summary>This shows up as a delete and create in the diff, so here's a <code>tree</code> view instead.</summary>
<p>

image

</p>
</details> 

<h1><em></em></h1> 

<details><summary>Symlinks are already used for <code>libhdf5</code>, and I have verified the resulting packages to be complete and working for the Vim packages.
</summary>
<p>

image

</p>
</details> 

I plan to apply the same patch deduplication to `emacs` and `emacs-x` in a later PR.

### CODE REVIEW

File: packages/vim-python/fix-writing-archive.patch
Comment (TomJo2000) [inline packages/vim-python/fix-writing-archive.patch:1]: If you take a look at the raw diff[^1] you will notice these "newly created" files are being set to filemode 120000[^2] which is a symlink.
[^1]: https://github.com/termux/termux-packages/pull/21540.diff
[^2]: https://git-scm.com/book/sv/v2/Git-Internals-Git-Objects#:~:text=120000

**Comment by fornwall** [comment]
Nice to get rid of the duplication of patches!

**Comment by Biswa96** [comment]
I had suggested a different approach. Instead of copying patches, how about applying the patches manually with `patch` command? For example, ncdu2 applies ncurses patches in build script.

**Comment by TomJo2000** [comment]
> I had suggested a different approach. Instead of copying patches, how about applying the patches manually with `patch` command? For example, ncdu2 applies ncurses patches in build script.

Yeah we had this discussion shortly in the discord.
Let me lay out my position on it a bit more clearly here.

I think the manual patching works great for `ncdu2` since we're pulling in patches from a loosely related package.
https://github.com/termux/termux-packages/blob/87f25d3eaa5adb4edf6a3e519c36ad92f40c4f61/packages/ncdu2/build.sh#L21-L24

I don't really like it in this case because it somewhat obscures that are are patches being applied to the package, and in the case of `vim-python` and `vim-gtk` those patches are exactly the same as for `vim`.

With symlinks you can instantly see, okay we have these patches, and they're coming from `../../packages/vim/*.patch`

**Pull request merged**